server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1059,6 +1077,42 @@ public void onFailure(Exception ex) {
         }
     }
 
+    private void releaseResourceLockFiles (
+        Collection<SnapshotId> snapshotIds,
+        RepositoryData repositoryData,
+        RemoteStoreLockManagerFactory remoteStoreLockManagerFactory,
+        ActionListener<Void> onAllShardsCompleted
+    ){
+        for (SnapshotId snapshotId: snapshotIds) {
+            List<String> indices = this.getSnapshotInfo(snapshotId).indices();
+            List<IndexId> indexIds = repositoryData.resolveIndices(indices);
+            for (IndexId indexId: indexIds) {
+                try {
+                    IndexMetadata indexMetadata = this.getSnapshotIndexMetaData(repositoryData,
+                        snapshotId, indexId);
+                    if (this.getSnapshotInfo(snapshotId).isRemoteStoreInteropEnabled()
+                        && indexMetadata.getSettings().getAsBoolean(
+                        IndexMetadata.SETTING_REMOTE_STORE_ENABLED,
+                        false)) {
+                        int numberOfShards = indexMetadata.getNumberOfShards();
+                        for (int shardId = 0; shardId < numberOfShards; shardId++) {
+                            final int finalShardId = shardId;
+                            String indexUUID = indexMetadata.getIndexUUID();
+                            String remoteStoreRepoForIndex = indexMetadata.getSettings().get(IndexMetadata.SETTING_REMOTE_STORE_REPOSITORY);
+                            RemoteStoreMetadataLockManager remoteStoreMetadataLockManager = remoteStoreLockManagerFactory
```

Comment:
As mentioned in above comment, this operation will fit well in the place where per shard operation is performed. That code is also fetching `IndexMetadata` which can be reused for this use case.

Reply:
Moved cleanup to separate method